src/main/java/net/irisshaders/iris/shaderpack/parsing/BooleanParser.java

```diff
@@ -0,0 +1,150 @@
+package net.irisshaders.iris.shaderpack.parsing;
+
+import net.irisshaders.iris.Iris;
+import net.irisshaders.iris.shaderpack.option.values.OptionValues;
+
+import java.util.EmptyStackException;
+import java.util.Stack;
+
+public class BooleanParser {
+	private enum OPERATION {
+		AND {
+			@Override
+			boolean compute(boolean value, Stack<Boolean> valueStack) {
+				return value & valueStack.pop();
```

Comment:
is there a reason for using bitwise instead of logical operators here?

Reply:
I could reverse the order then it doesn't matter